test_utils_helpers.py

```
@@ -0,0 +1,1451 @@
+    @pytest.mark.slow
+    def test_memory_usage_large_operations(self):
+        """Test memory usage with large data operations"""
+        # Test with large data structures
+        large_nested_dict = {}
+        current_level = large_nested_dict
+        
+        # Create 50 levels of nesting with data at each level
+        for i in range(50):
+            current_level[f"level_{i}"] = {
+                "data": [f"item_{j}" for j in range(100)],  # 100 items per level
+                "metadata": {"level": i, "timestamp": time.time()},
```

Ruff (0.11.9)

Local variable `time` referenced before assignment

(F823)